sql/core/src/main/scala/org/apache/spark/sql/execution/joins/SortMergeJoinExec.scala

```diff
@@ -680,13 +680,36 @@ case class SortMergeJoinExec(
       ctx.currentVars = streamedVars ++ bufferedVars
       val cond = BindReferences.bindReference(
         condition.get, streamedPlan.output ++ bufferedPlan.output).genCode(ctx)
-      // evaluate the columns those used by condition before loop
+      // Evaluate the columns those used by condition before loop
       val before =
         s"""
            |boolean $loaded = false;
            |$streamedBefore
          """.stripMargin
 
+      val loadStreamed =
+        s"""
+           |if (!$loaded) {
+           |  $loaded = true;
+           |  $streamedAfter
+           |}
+         """.stripMargin
```

Comment:
nit: seems `loaded` is not needed for `LeftAnti` case.

Reply:
`loadStreamed` is not used by `LeftAnti`.
I think you are referring to `boolean $loaded = false;` in `before` should not be needed, right?

Reply:
yea, looks like for `LeftAnti`, it doesn't rely on `loaded` to do `streamedAfter`.

Reply:
Created https://github.com/apache/spark/pull/32681 as followup, thanks.